lib/ret/entity_manager.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Ret.EntityManager do
+  alias Ecto.Multi
+  alias Ret.{Entity, SubEntity, Repo}
+  import Ecto.Query
+
+  def create_entity(hub, %{updates: updates} = entity_params) do
+    Multi.new()
+    |> Multi.insert(:entity, Entity.changeset(%Entity{}, hub, entity_params))
+    |> Multi.insert_all(
+      :sub_entities,
+      SubEntity,
+      fn %{entity: entity} ->
+        Enum.map(updates, &SubEntity.for_bulk_insert(hub, entity, &1))
+      end
+    )
```

Comment:
I’m fairly certain this whole `create_entity/2` function is achievable with a single `Repo.insert/1` sans the `Multi`. `Ecto.Changeset.cast_assoc/3` is the key. We can chat.